docs/proposals/hpa/cronfederatedhpa.md

```diff
@@ -0,0 +1,339 @@
+nitty-gritty.
+-->
+
+### User Stories(Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As a cluster administrator, I know that there will be a sudden peak traffic every days at 9 o'clock AM. So I want to scale up the related service in advance(like 30 minutes) to meet the sharp load peaks, ensure the service is available when the time comes. Also, after the sharp load peaks have gone(like 2 hours latter), I want to scale down the related service to save cloud costs.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go into as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate them? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+`CronFederatedHPA` could scale FederatedHPA, but also scale the workloads directly. 
+So if you can set `CronFederatedHPA` to scale the workloads, which also will be scaled by with `FederatedHPA`, it will have conflicts as follows:  
+<image src="./statics/cronfederatedhpa-constraints.png" width="300px">
+
+## Design Details
+### Architecture
+<image src="./statics/cronfederatedhpa-architecture.png" width="300px">
+
+From the above architecture diagram, CronFederatedHPA should do following things:
+* `CronFederatedHPA` allow users to scale the workloads at specific time.
+* `CronFederatedHPA` allow users to set the `FederatedHPA` at specific time.
+
+### API Definition
+In order to satisfy the above requirements, we defined the API as following:
+```go
+type CronFederatedHPA struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of the CronFederatedHPASpec.
+	// +required
+	Sepc CronFederatedHPASpec `json:"spec"`
+
+	// Status is the current status of the CronFederatedHPA.
+	// +optional
+	Status CronFederatedHPAStatus `json:"status"`
+}
+
+type CronFederatedHPASpec struct {
+	// ScaleTargetRef points to the target resource to scale.
+	// It could be FederatedHPA, Deployment, StatefulSet, or other resource with scale sub-resource.
+	// +required
+	ScaleTargetRef ScaleTargetRef `json:"scaleTargetRef"`
+
+	// Rules is the list of CronFederatedHPARule,
+	// it define when and how to scale the related resources.
+	Rules []CronFederatedHPARule `json:"rules"`
+}
+
+type CronFederatedHPARule struct {
+	// RuleName is the name of the CronFederatedHPARule.
+	// +required
+	RuleName string `json:"ruleName"`
```

Comment:
Should we restrict the name like what we did on [AffinityName](https://github.com/karmada-io/karmada/blob/5e14c5a3145c1089a0b56f4259477f6ee9cdf444/pkg/apis/policy/v1alpha1/propagation_types.go#LL345C1-L346C1)?